packages/shared/src/components/MainLayout.tsx

```diff
@@ -103,18 +124,62 @@ export default function MainLayout({
   enableSearch,
   onShowDndClick,
 }: MainLayoutProps): ReactElement {
-  const { user, loadingUser } = useContext(AuthContext);
+  const router = useRouter();
```

Comment:
I think we should consider taking out everything auth related from here.
It already has quite a lot of code here

Reply:
I have now completely extracted auth-related code blocks from the MainLayout.